src/renderer/components/+extensions/extensions.tsx

```diff
@@ -308,22 +303,77 @@ async function installOnDrop(files: File[]) {
   await requestInstalls(files.map(({ path }) => path));
 }
 
+async function requestInstallVersion(url: URLParse): Promise<string | null> {
+  const disposer = ExtensionInstallationStateStore.startPreInstall();
+  let json: Record<string, any>;
+
+  try {
+    const ticket = downloadFile({ url, gzip: false, timeout: 60_000 });
+    const res = await ticket.promise;
+
+    json = JSON.parse(res.toString());
+  } catch (error) {
+    console.error(error);
+    Notifications.error("Failed to get registry information for that extension");
+    disposer();
+
+    return null;
+  }
+
+  let selected: string;
+  const options = Object.keys(json.versions)
+    .map(version => new SemVer(version, { loose: true, includePrerelease: true }))
+    .sort((a, b) => b.compare(a))
+    .map(a => a.format());
+
+  const removeNotification = Notifications.info(
```

Comment:
you're probably getting rid of this notification? If not when clicking the X to close the notification (cancel the install) the Install button stays dimmed and the spinner spins and spins...